newrelic/resource_newrelic_obfuscation_expression.go

```diff
@@ -0,0 +1,200 @@
+package newrelic
+
+import (
+	"context"
+	"errors"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/newrelic/newrelic-client-go/v2/newrelic"
+	_ "github.com/newrelic/newrelic-client-go/v2/newrelic"
+	nrErrors "github.com/newrelic/newrelic-client-go/v2/pkg/errors"
+	"github.com/newrelic/newrelic-client-go/v2/pkg/logconfigurations"
+)
+
+//
+func resourceNewRelicObfuscationExpression() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNewRelicObfuscationExpressionCreate,
+		ReadContext:   resourceNewRelicObfuscationExpressionRead,
+		UpdateContext: resourceNewRelicObfuscationExpressionUpdate,
+		DeleteContext: resourceNewRelicObfuscationExpressionDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:        schema.TypeInt,
+				Description: "The account id associated with the obfuscation expression.",
+				Computed:    true,
+				Optional:    true,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "Description of expression.",
+				Optional:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "Name of expression.",
+				Required:    true,
+			},
+			"regex": {
+				Type:        schema.TypeString,
+				Description: "Regex of expression.",
+				Required:    true,
+			},
+			"expression_id": {
```

Comment:
We can remove this from the schema and just use the ID from Terraform. Customers find it confusing when there are too many IDs.